deep_reference_parser/prodigy/reference_to_token_annotations.py

```diff
@@ -10,31 +10,54 @@
 
 
 class TokenTagger:
-    """
-    Converts data in prodigy format with full reference spans to per-token spans
+    def __init__(self, task="splitting", lowercase=True):
+        """
+        Converts data in prodigy format with full reference spans to per-token
+            spans
 
-    Expects one of four lables for the spans:
+        Args:
+            task (str): One of ["parsing", "splitting"]. See below further
+                explanation.
+            lowercase (bool): Automatically convert upper case annotations to
+                lowercase under the parsing scenario.
 
-    * BE: A complete reference
-    * BI: A frgament of reference that captures the beginning but not the end
-    * IE: A frgament of reference that captures the end but not the beginning
-    * II: A fragment of a reference that captures neither the beginning nor the
-        end .
-    """
+        Since the parsing, splitting, and classification tasks have quite
+        different labelling requirements, this class behaves differently
+        depending on which task is specified in the task argument.
+
+        For splitting:
 
-    def __init__(self):
+        Expects one of four labels for the spans:
+
+        * BE: A complete reference
+        * BI: A frgament of reference that captures the beginning but not the end
+        * IE: A frgament of reference that captures the end but not the beginning
+        * II: A fragment of a reference that captures neither the beginning nor the
+            end .
+
+        Depending on which label is applied the tokens within the span will be
+        labelled differently as one of ["b-r", "i-r", "e-r", "o"].
+
+        For parsing:
+
+        Expects any arbitrary label for spans. All tokens within that span will
+        be labelled with the same span.
```

Comment:
can you give some more information about the possible labels. ["b-a", "i-a", "e-a", ..] is that for author?